nmostesting/suites/IS1102Test.py

```diff
@@ -13,16 +13,1402 @@
+
+        for sender_id in self.flow_format_audio:
+            _, response = TestHelper.do_request(
+                "GET",
+                self.compat_url + "senders/" + sender_id + "constraints/supported/ ",
+            )
+            if response.status_code != 200:
+                return test.FAIL(response.json())
+            supportedConstraints = response.json()["parameter_constraints"]
+            for item in supportedConstraints:
+                if not re.search(pattern, item):
+                    return test.FAIL("only x-nmos:cap constraints are allowed")
+            for item in REF_SUPPORTED_CONSTRAINTS_AUDIO:
+                if item not in supportedConstraints:
+                    return test.FAIL(item + "is not in supportedConstraints")
+        return test.PASS()
+
+    def test_02_02_04_01(self, test):
+        """
+        Verify that changing the constraints of an
+        IS-11 sender(video) changes the version of
+        the associated IS-04 sender.
+        """
+        if len(self.flow_format_video) == 0:
+            return test.UNCLEAR("There is no video format")
+
+        for sender_id in self.flow_format_video:
+            _, response = TestHelper.do_request(
+                "GET", self.node_url + "senders/" + sender_id
+            )
+            if response.status_code != 200:
+                return test.FAIL(response.json())
+            version = response.json()["version"]
+            self.version[sender_id] = version
+            self.grain_rate_constraints[sender_id] = {
+                "constraint_sets": [
+                    {
+                        "urn:x-nmos:cap:format:grain_rate": {
+                            "enum": [self.flow_grain_rate[sender_id]]
+                        }
+                    }
+                ]
+            }
+            self.empty_constraints[sender_id] = {"constraint_sets": []}
+            _, response = TestHelper.do_request(
+                "PUT",
+                self.compat_url + "senders/" + sender_id + "constraints/active/",
+                json=self.grain_rate_constraints[sender_id],
+            )
+            if response.status_code != 200:
+                return test.FAIL(response.json())
+            _, response = TestHelper.do_request(
+                "GET", self.node_url + "senders/" + sender_id
+            )
+            if response.status_code != 200:
+                return test.FAIL(response.json())
+            version = response.json()["version"]
+            if version == self.version[sender_id]:
+                return test.FAIL()
+            self.version[sender_id] = version
+            _, response = TestHelper.do_request(
+                "GET", self.compat_url + "senders/" + sender_id + "constraints/active/ "
+            )
+            if response.status_code != 200:
+                return test.FAIL(response.json())
+            constraints = response.json()
+            if constraints != self.grain_rate_constraints[sender_id]:
```

Comment:
Cannot do a pure string compare because the constraint set object (https://raw.githubusercontent.com/AMWA-TV/bcp-004-01/v1.0.x/APIs/schemas/constraint_set.json) have optional meta:preference and meta:enabled properties that have a default value. Because of those defaults the Node could return active constraints with those additional properties which is equivalent to the PUT ones ... Perform a comparison using the schema rules. ... Also note that the constraints may be returned in any ordering that is legal by the schema.